crates/forge/src/test_execution_syscall_handler.rs

```diff
@@ -266,20 +269,56 @@ impl TestExecutionSyscallHandler<'_, '_> {
-                self.contract_execution_syscall_handler
-                    .cheatable_syscall_handler
-                    .cheatnet_state
-                    .start_warp(contract_address, value);
+                // First felt encodes the variant
+                if inputs[0] == Felt252::from(0) {
+                    self.contract_execution_syscall_handler
+                        .cheatable_syscall_handler
+                        .cheatnet_state
+                        .start_warp(StartWarpTarget::All(inputs[1].clone()));
+                } else if inputs[0] == Felt252::from(1) {
+                    let warp_target =
+                        StartWarpTarget::One((inputs[1].to_contract_address(), inputs[2].clone()));
+                    self.contract_execution_syscall_handler
+                        .cheatable_syscall_handler
+                        .cheatnet_state
+                        .start_warp(warp_target);
+                } else if inputs[0] == Felt252::from(2) {
+                    let warp_tuples: Vec<_> = inputs[2..]
+                        .chunks_exact(2)
+                        .map(|chunk| (chunk[0].to_contract_address(), chunk[1].clone()))
+                        .collect();
+                    self.contract_execution_syscall_handler
+                        .cheatable_syscall_handler
+                        .cheatnet_state
+                        .start_warp(StartWarpTarget::Multiple(warp_tuples));
+                }
+
                 Ok(())
             }
             "stop_warp" => {
-                let contract_address = inputs[0].to_contract_address();
-                self.contract_execution_syscall_handler
-                    .cheatable_syscall_handler
-                    .cheatnet_state
-                    .stop_warp(contract_address);
+                // First felt encodes the variant
+                if inputs[0] == Felt252::from(0) {
+                    self.contract_execution_syscall_handler
+                        .cheatable_syscall_handler
+                        .cheatnet_state
+                        .stop_warp(StopWarpTarget::All);
+                } else if inputs[0] == Felt252::from(1) {
+                    let warp_target = StopWarpTarget::One(inputs[1].to_contract_address());
+                    self.contract_execution_syscall_handler
+                        .cheatable_syscall_handler
+                        .cheatnet_state
+                        .stop_warp(warp_target);
+                } else if inputs[0] == Felt252::from(2) {
+                    let warp_targets: Vec<_> = inputs[2..]
+                        .iter()
+                        .map(Felt252::to_contract_address)
+                        .collect();
+                    self.contract_execution_syscall_handler
+                        .cheatable_syscall_handler
+                        .cheatnet_state
+                        .stop_warp(StopWarpTarget::Multiple(warp_targets));
+                }
```

Comment:
`match` statement would be more idiomatic, look at the way `"spy_events"` arm is implemented

Reply:
I had to use match guards. Let me know if it looks okay.